dhall/doctest/Main.hs

```diff
@@ -38,6 +46,11 @@ main = do
 
             -- , prefix </> "src"
             , "-i" <> (prefix </> "src")
+#ifdef MIN_VERSION_GLASGOW_HASKELL
+#if MIN_VERSION_GLASGOW_HASKELL(8,6,1,0)
```

Comment:
I don't think I've ever seen these pragmas before. Why not use the conventional

```
#if __GLASGOW_HASKELL__ >= 806
```

instead?

Reply:
Found it [here](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/phases.html#standard-cpp-macros). I used that one because DerivingVia was introduced in a patchlevel version (per [this](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#deriving-via)), and I thought this looked ~better~ less bad than nesting `__GLASGOW_HASKELL__ ` and `__GLASGOW_HASKELL_PATCHLEVEL1__`

Reply:
Oh! I think you'll find that pretty much every extension was added in minor version X.Y.1, but that's only because 1 is the starting minor version number for official releases. Just

     #if __GLASGOW_HASKELL__ >= 806

should be fine.

Reply:
Ah! That makes sense, thanks!

- [ ] replace pragma